CMakeLists.txt

```diff
@@ -547,14 +547,17 @@ else ()
 endif ()
 
 if ( NOT DISABLE_SPICY )
+    set(USE_SPICY_ANALYZERS yes)
+
     if ( NOT SPICY_ROOT_DIR )
+        set(HAVE_SPICY yes) # evaluated by Spicy plugin build
```

Comment:
We never set a value for this for the `else` branch.

This makes me wonder, do we actually need `HAVE_SPICY` on top of `NOT DISABLE_SPICY`? Outside of `NOT DISABLE_SPICY AND SPICY_ROOT_DIR` it seems to mirror `NOT DISABLE_SPICY` perfectly; if the value for the missing branch is `OFF` they would be equivalent.

Reply:
This is how it used to be, and actually it's exactly what I broke originally when the external builds weren't working anymore: I had changed it to get set it on all branches, so that it could be used like `USE_SPICY_ANALYZERS` is used now.  The problem with that is that the spicy-plugin picks up on `HAVE_SPICY` being set; if it is, it doesn't look for an external Spicy installation anymore. So the trick is to leave it undefined if we want it to do that. 

Yeah, it's messy, but nothing I wanted to touch further right now ...